slasher/src/database/redb_impl.rs

```diff
@@ -0,0 +1,281 @@
+#![cfg(feature = "redb")]
+use crate::{
+    database::{
+        interface::{Key, OpenDatabases, Value},
+        *,
+    },
+    Config, Error,
+};
+use derivative::Derivative;
+use redb::{ReadableTable, TableDefinition};
+use std::{borrow::Cow, path::PathBuf};
+
+const BASE_DB: &str = "slasher_db";
+
+#[derive(Debug)]
+pub struct Environment {
+    _db_count: usize,
+    db: redb::Database,
+}
+
+#[derive(Debug)]
+pub struct Database<'env> {
+    table_name: String,
+    _phantom: PhantomData<&'env ()>,
+}
+
+#[derive(Derivative)]
+#[derivative(Debug)]
+pub struct RwTransaction<'env> {
+    #[derivative(Debug = "ignore")]
+    txn: redb::WriteTransaction,
+    _phantom: PhantomData<&'env ()>,
+}
+
+#[derive(Derivative)]
+#[derivative(Debug)]
+pub struct Cursor<'env> {
+    #[derivative(Debug = "ignore")]
+    txn: &'env redb::WriteTransaction,
+    db: &'env Database<'env>,
+    current_key: Option<Cow<'env, [u8]>>,
+}
+
+impl Environment {
+    pub fn new(config: &Config) -> Result<Environment, Error> {
+        let db_path = match config.database_path.join(BASE_DB).as_path().to_str() {
```

Comment:
do we definitely need to join the `BASE_DB` here? I think the caller takes care of that already when constructing the `config.database_path`